mlflow/dspy/load.py

```diff
@@ -15,28 +14,23 @@
 
 _DEFAULT_MODEL_PATH = "data/model.pkl"
 
-_logger = logging.getLogger(__name__)
 
-
-def _set_tracer_context(model_path, callbacks):
+def _set_dependency_schema_to_tracer(model_path, callbacks):
     """
-    Set dependency schemas from the saved model metadata to the tracer's prediction context.
+    Set dependency schemas from the saved model metadata to the tracer
+    to propagate it to inference traces.
     """
     from mlflow.dspy.callback import MlflowCallback
 
     tracer = next((cb for cb in callbacks if isinstance(cb, MlflowCallback)), None)
     if tracer is None:
         return
-    context = tracer._prediction_context
 
     model = Model.load(model_path)
-    if schema := _get_dependencies_schema_from_model(model):
-        context.update(**schema)
+    tracer._dependencies_schema = _get_dependencies_schema_from_model(model)
```

Comment:
nit: it might be better if we define a setter for _dependencies_schema rather than manipulating private field outside of the class.